static/js/lib/ckeditor/plugins/CustomLink.ts

```diff
@@ -95,14 +190,47 @@ export default class CustomLink extends Plugin {
     }
   }
 
-  // This function can contain your custom logic to generate a new href
+  /**
+   * Generate a resource link href from a resource ID.
+   *
+   * This is a simple wrapper around ResourceLinkMarkdownSyntax.makeResourceLinkHref()
+   * that creates properly formatted resource links with the correct URL structure
+   * and query parameters needed by the OCW system.
+   *
+   * @param resourceID - The text_id of the external resource content item
+   * @returns A properly formatted resource link href
+   */
   _getResourceLink(resourceID: string): string {
     // Example: Appending a query parameter to the original href
     return this.syntax.makeResourceLinkHref(resourceID)
   }
 }
 
-async function getExternalResource(
+/**
+ * Create an external resource content item via the OCW Studio API.
+ *
+ * This function handles the core logic of external resource creation:
+ *
+ * 1. DOMAIN ANALYSIS: Determines if the URL is external by comparing its hostname
+ *    to SETTINGS.sitemapDomain. Same-domain URLs get no license warning.
```

Comment:
Again, it seems to me that all URLs are external resources. E.g., https://ocw-studio-rc.odl.mit.edu/sites/resstr-001-geographic-information-system-gis-tutorial/type/external-resource/edit/134ee4ce-749a-46e3-9baa-f1f062a1bce6/ 

So we don't determine external vs non-external via domain, we determine "needs content warning" vs "does not need content warning".